src/kernel/kmain.zig

```diff
@@ -2,17 +2,21 @@
 
 const std = @import("std");
 const builtin = @import("builtin");
-const arch = if (builtin.is_test) @import("../../test/kernel/arch_mock.zig") else @import("arch.zig").internals;
+const arch = @import("arch.zig").internals;
 const multiboot = @import("multiboot.zig");
 const tty = @import("tty.zig");
 const vga = @import("vga.zig");
 const log = @import("log.zig");
 const serial = @import("serial.zig");
-const mem = @import("mem.zig");
+const mem = if (builtin.is_test) @import("mocking").mem else @import("mem.zig");
+
+// This is for unit testing as we need to export KERNEL_ADDR_OFFSET as it is no longer available
+// from the linker script
+export var KERNEL_ADDR_OFFSET: u32 = if (builtin.is_test) 0xC0000000 else undefined;
```

Comment:
Will this override the KERNEL_ADDR_OFFSET set by the linker script?

Reply:
Will probs remove now as I am re-adding the constants.zig file